syntax = "proto3";

package NPersQueue.NErrorCode;
option java_package = "com.yandex.ydb.persqueue";

enum EErrorCode {
    OK = 0;
    INITIALIZING = 1;
    OVERLOAD = 2;
    BAD_REQUEST = 3;
    WRONG_COOKIE = 4;
    SOURCEID_DELETED = 24;

    WRITE_ERROR_PARTITION_IS_FULL = 5;
    WRITE_ERROR_DISK_IS_FULL = 15;
    WRITE_ERROR_BAD_OFFSET = 19;

    CREATE_SESSION_ALREADY_LOCKED = 6;
    DELETE_SESSION_NO_SESSION = 7;

    READ_ERROR_IN_PROGRESS = 8;
    READ_ERROR_NO_SESSION = 9;
    READ_TIMEOUT = 10; // TODO: move to pqlib codes
    READ_ERROR_TOO_SMALL_OFFSET = 11;
    READ_ERROR_TOO_BIG_OFFSET = 12;

    SET_OFFSET_ERROR_COMMIT_TO_FUTURE = 13;
    SET_OFFSET_ERROR_COMMIT_TO_PAST = 25;

    TABLET_IS_DROPPED = 14;

    READ_NOT_DONE = 16;

    UNKNOWN_TOPIC = 17;

    ACCESS_DENIED = 18;
    CLUSTER_DISABLED = 20;

    WRONG_PARTITION_NUMBER = 21;

    CREATE_TIMEOUT = 22; // TODO: move to pqlib codes
    IDLE_TIMEOUT = 23; // TODO: move to pqlib codes

    PREFERRED_CLUSTER_MISMATCHED = 26;

    TABLET_PIPE_DISCONNECTED = 27;

    SCHEMA_ERROR = 28;

    ERROR = 100;
}
